examples/basic.rs

```diff
@@ -16,17 +16,27 @@ fn main() {
 
     let mut reader_id = event_handler.register_reader();
 
-
     // Should be empty, because reader was created after the write
-    assert_eq!(Ok([].to_vec()), event_handler.read(&mut reader_id));
+    assert_eq!(
+        Vec::<TestEvent>::default(),
+        event_handler
+            .read(&mut reader_id)
+            .unwrap()
+            .cloned()
+            .collect::<Vec<_>>()
```

Comment:
Is this type annotation necessary?

Reply:
Yes, because otherwise the collect is undefined as to the type of the collection, and gives a compile error.

Reply:
I thought it can infer that you need `Vec` since you comare it with another `Vec`.

Reply:
It can't ..